src/components/main/productDetails/ProductDetails.ts

```diff
@@ -0,0 +1,153 @@
+      }
+      window.location.hash = `#${Pages.Cart}`;
+      setTimeout(() => Payment.render(), 500);
+    });
+
+    return header;
+  }
+
+  private addToCart(buttonAddToCart: HTMLElement): void {
+    if (buttonAddToCart.classList.length === 1) {
+      buttonAddToCart.textContent = 'REMOVE FROM CART';
+    } else {
+      buttonAddToCart.textContent = 'ADD TO CART';
+    }
+    Product.addProductToCart(buttonAddToCart, this.product, 'product-details__button_remove');
+  }
+
+  public renderDescription(): HTMLElement {
+    const productDetails: HTMLElement = document.createElement('div');
+    productDetails.classList.add('product-details__description');
+
+    const productTitle: HTMLElement = document.createElement('div');
+    productTitle.classList.add('product-details__title');
+    productTitle.innerText = `${this.product.titleDetail}`;
+    productDetails.append(productTitle);
+
+    const productReting: HTMLElement = document.createElement('div');
+    productReting.classList.add('product-details__rating');
+    const ratingStars: string = Product.fillRating(this.product.rating);
+    productReting.innerHTML = `
+      <div class="product__rating">${ratingStars}</div>
+      <span class="product__rating-text">${this.product.rating}</span>
+      </div>
+    `;
+    productDetails.append(productReting);
+
+    productDetails.append(this.price()[0]);
+    productDetails.append(this.price()[1]);
+
+    const productBrand: HTMLElement = document.createElement('div');
+    productBrand.classList.add('product-details__article');
+    productBrand.innerHTML = `<span>Brand:</span> ${this.product.brand}`;
+    productDetails.append(productBrand);
+
+    const productCategory: HTMLElement = document.createElement('div');
+    productCategory.classList.add('product-details__article');
+    productCategory.innerHTML = `<span>Category:</span> ${this.product.category}`;
+    productDetails.append(productCategory);
+
+    const productStock: HTMLElement = document.createElement('div');
+    productStock.classList.add('product-details__article');
+    productStock.innerHTML = `<span>Stock:</span> ${this.product.stock}`;
+    productDetails.append(productStock);
+
+    const productDescription: HTMLElement = document.createElement('ul');
+    productDescription.innerText = 'Description:';
+    productDescription.classList.add('product-details__specification');
+    productDetails.append(productDescription);
+
+    this.product.description.forEach((el) => {
+      const listItem: HTMLElement = document.createElement('li');
+      listItem.classList.add('product-details__specification-item');
+      listItem.innerText = el;
+      productDescription.append(listItem);
+    });
+
+    return productDetails;
+  }
+
+  public renderGallery(): HTMLElement {
+    const productGallery: HTMLElement = document.createElement('div');
+    productGallery.classList.add('product-details__gallery');
+
+    new Gallery(productGallery, this.product.images);
```

Comment:
What is the usage of instantiating a class without assigning it to any variable? I guess you move all logic of the `Gallery` class inside its constructor. That's completely wrong, you should better create a method for combining all the class actions (like `render` or `start`)